pagopa/io-web-profile
# [IOPID-1225] [IOPID-1229] [IOPID-1238] : Login FLow

## Short description

The purpose of this PR is to address the following issues:

- [IOPID-1225]
- [IOPID-1229]
- [IOPID-1238]

## List of changes proposed in this pull request

To achieve this, we have refactored the getLoginFlow method to handle the page to be rendered after login, using the 'login page route info' that was used for the login rather than the spidLevel. 
This change is necessary because during the Logic with CIE, users can log in with L1 or L2, and we cannot determine the level used by the user before the login. 
This inconsistency causes issues with the logout flow. By using the login page as a reference, we resolve the rendering differences.

## How to test

Run the project and test the various cases reported. 
Note that in the local environment, we are unable to simulate LOGIN CIE provider, so we have used the hubspidlogin editing on goCIE() method (idps.ts) using the hubspid login default login path.  So for this please edit line 139 (idps.ts) to 

```
  window.location.assign(
    `${process.env.NEXT_PUBLIC_URL_SPID_LOGIN}?entityID=xx_testenv2&authLevel=SpidL1&RelayState=ioapp`
  );
```
If you want to Login with different level change the authLevel param

FLOW (logout):

Login via SPID, CIE (LEVEL1 and LEVEL2): After login, you should land on the logout session page as reported in:
- https://pagopa.atlassian.net/browse/IOPID-1225
- https://pagopa.atlassian.net/browse/IOPID-1229

FLOW (loginL3):
From the card, unlock the profile, click -> "discover how to do it". You will be redirected to the page loginL3. 
Perform the CIE login and verify that you are redirected to the correct page after login, as reported in:
- https://pagopa.atlassian.net/browse/IOPID-1238

[IOPID-1225]: https://pagopa.atlassian.net/browse/IOPID-1225?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[IOPID-1229]: https://pagopa.atlassian.net/browse/IOPID-1229?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[IOPID-1238]: https://pagopa.atlassian.net/browse/IOPID-1238?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
